docs/shared/recipes/generators/register-sync-generator.md

```diff
@@ -0,0 +1,110 @@
+# Register a Sync Generator
+
+Sync generators are intended to keep applicable files in sync with your project graph before a task is run. From a technical perspective, a sync generator is no different from any other generator, but it has some additional performance considerations and needs to be registered in a particular way.
```

Comment:
I'm unsure what would be the best way to phrase it, but sync generators are not limited to keeping files in sync with the project graph.